charms/slurmctld/src/charm.py

```diff
@@ -127,7 +144,42 @@ def _on_install(self, event: InstallEvent) -> None:
             logger.error(e.message)
             event.defer()
 
-        self._on_write_slurm_conf(event)
+    def _on_start(self, event: StartEvent) -> None:
+        """Set cluster_name and write slurm.conf.
+
+        Note: The start hook can execute multiple times in a charms lifecycle,
+              for example, after a reboot of the underlying instance. This code safeguards
+              against the potentiality of changing the cluster_name in subsequent start hook
+              executions by applying logic that ensures the cluster_name is only set on the
+              first execution of this hook, we log and return on any subsequent start hook
+              event executions.
+        """
+        if self.unit.is_leader():
+            if self._slurmctld_peer.cluster_name is None:
+                if (charm_config_cluster_name := str(self.config.get("cluster-name", ""))) != "":
+                    cluster_name = charm_config_cluster_name
+                else:
+                    cluster_name = f"{CLUSTER_NAME_PREFIX}-{generate_random_string(4)}"
```

Comment:
Could we just replace `generate_random_string(...)` with `token_urlsafe`? That way we don't need an external dependency like `petname` or have to incur any technical debt from carrying our own custom module:

```suggestion
                    cluster_name = f"{CLUSTER_NAME_PREFIX}-{secrets.token_urlsafe(3)}"
```

_`token_urlsafe(3)` generates 4 characters:_

```pyi
>>> CLUSTER_NAME_PREFIX = "charmed-hpc"
>>> f"{CLUSTER_NAME_PREFIX}-{secrets.token_urlsafe(3)}"
'charmed-hpc-Nt9c'
```